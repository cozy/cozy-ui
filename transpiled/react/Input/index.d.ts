export default Input;
import Input from "@material-ui/core/Input";
