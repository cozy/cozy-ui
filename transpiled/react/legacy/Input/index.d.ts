export default Input;
declare const Input: React.ForwardRefExoticComponent<React.RefAttributes<any>>;
import React from "react";
