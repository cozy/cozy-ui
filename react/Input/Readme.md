### Input's available types (text, password, email, url)

```
<form>
  <p><Input placeholder="This is a input[type=text]" /></p>
  <p><Input type="password" placeholder="This is a input[type=password]" /></p>
  <p><Input type="email" placeholder="This is a input[type=email]" /></p>
  <p><Input type="url" placeholder="This is a input[type=url]" /></p>
  <p><Input type="number" placeholder="This is a input[type=number]" min="10" max="100" step="10" /></p>
</form>
```

### Disabled Input

```
<form>
  <Input disabled value="I'm disabled" />
</form>
```

### Input when there's an error

```
<Input error placeholder="This is a input[type=text] with error" />
```

### Alternative input sizes

By default, the size is `large`.

```
<div>
  <p>
    <Input placeholder="I have a tiny size" size="tiny" />
  </p>
  <p>
    <Input placeholder="I have a medium size" size="medium" />
  </p>
</div>
```

### Full width inputs

```
<Input placeholder="I'm full width" fullwidth />
```

### inputRef ( set focus / blur / ... programmatically)

If you need to programmatically access the underlying `<input />` for example to give focus or move the caret, you can use the `inputRef` prop, that is passed to the `ref` property of the `<input />`.

```

class InputComponent extends React.Component {
  constructor() {
    super()
    this.component = null;
    this.setFocus = this.setFocus.bind(this)
  }
  setFocus() {
    this.component.focus()
  }
  render() {
    return (
      <div>
        <Button onClick={this.setFocus}>Set Focus</Button>
        <Input inputRef={c => this.component = c} />
      </div>
    )
  }
}
;<InputComponent />
```

### Props forwarding

`Input` forwards unknown props to the underlying `<input />` element.

```
<div>
  <Input placeholder='Type to see changes' value={state.value} onChange={ev => setState({value: ev.target.value})} />&nbsp;&nbsp;
  Value: { state.value }
</div>
```
