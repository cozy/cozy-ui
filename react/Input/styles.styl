@require '../../stylus/components/forms'

.c-input-text
    @extend $input-text

.c-input-text--tiny
    @extend $input-text--tiny

.c-input-text--medium
    @extend $input-text--medium

.c-input-text--large
    @extend $input-text--large

.c-input-text--fullwidth
    @extend $input-text--fullwidth
