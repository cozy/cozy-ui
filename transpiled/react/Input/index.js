import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["autoComplete", "disabled", "type", "id", "className", "value", "placeholder", "error", "size", "fullwidth"];
import React, { forwardRef } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
var styles = {
  "c-input-text": "styles__c-input-text___3TAv1",
  "is-error": "styles__is-error___3lsCJ",
  "c-input-text--tiny": "styles__c-input-text--tiny___MzMoD",
  "c-input-text--medium": "styles__c-input-text--medium___28jPV",
  "c-input-text--large": "styles__c-input-text--large___28EaR",
  "c-input-text--fullwidth": "styles__c-input-text--fullwidth___33o_f"
};
var Input = /*#__PURE__*/forwardRef(function (props, ref) {
  var _cx;

  var autoComplete = props.autoComplete,
      disabled = props.disabled,
      type = props.type,
      id = props.id,
      className = props.className,
      value = props.value,
      placeholder = props.placeholder,
      error = props.error,
      size = props.size,
      fullwidth = props.fullwidth,
      restProps = _objectWithoutProperties(props, _excluded);

  return /*#__PURE__*/React.createElement("input", _extends({
    "aria-disabled": disabled,
    autoComplete: autoComplete,
    type: type,
    id: id,
    ref: ref,
    className: cx(styles['c-input-text'], (_cx = {}, _defineProperty(_cx, styles["is-error"], error), _defineProperty(_cx, styles["c-input-text--".concat(size)], size), _defineProperty(_cx, styles["c-input-text--fullwidth"], fullwidth), _cx), className),
    disabled: disabled,
    placeholder: placeholder,
    value: value,
    spellCheck: false
  }, restProps));
});
Input.displayName = 'Input';
Input.propTypes = {
  autoComplete: PropTypes.string,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(['date', 'email', 'number', 'password', 'text', 'url', 'tel']),
  id: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  placeholder: PropTypes.string,
  error: PropTypes.bool,
  size: PropTypes.oneOf(['tiny', 'medium', 'large']),
  fullwidth: PropTypes.bool,
  onChange: PropTypes.func,
  onKeyUp: PropTypes.func
};
Input.defaultProps = {
  type: 'text',
  error: false,
  fullwidth: false,
  size: 'large'
};
export default Input;