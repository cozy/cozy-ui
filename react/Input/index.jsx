import React from 'react'
import cx from 'classnames'
import PropTypes from 'prop-types'
import styles from './styles.styl'

const Input = props => {
  const {
    autoComplete,
    disabled,
    type,
    id,
    className,
    value,
    placeholder,
    error,
    size,
    fullwidth,
    inputRef,
    ...restProps
  } = props
  return (
    <input
      aria-disabled={disabled}
      autoComplete={autoComplete}
      type={type}
      id={id}
      ref={inputRef}
      className={cx(
        styles['c-input-text'],
        {
          [styles[`is-error`]]: error,
          [styles[`c-input-text--${size}`]]: size,
          [styles[`c-input-text--fullwidth`]]: fullwidth
        },
        className
      )}
      disabled={disabled}
      placeholder={placeholder}
      value={value}
      {...restProps}
    />
  )
}

Input.propTypes = {
  autoComplete: PropTypes.string,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf([
    'date',
    'email',
    'number',
    'password',
    'text',
    'url',
    'tel'
  ]),
  id: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  placeholder: PropTypes.string,
  error: PropTypes.bool,
  size: PropTypes.oneOf(['tiny', 'medium', 'large']),
  fullwidth: PropTypes.bool,
  /**
   * Use that property to pass a ref callback to the native input component.
   */
  inputRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
  onChange: PropTypes.func,
  onKeyUp: PropTypes.func
}

Input.defaultProps = {
  type: 'text',
  error: false,
  fullwidth: false,
  size: 'large'
}

export default Input
