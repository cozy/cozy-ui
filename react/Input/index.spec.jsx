import React from 'react'

import Input from './index'

describe('Input component', () => {
  it('should support number type', () => {
    const component = shallow(
      <Input type="number" value="42" onChange={jest.fn()} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
