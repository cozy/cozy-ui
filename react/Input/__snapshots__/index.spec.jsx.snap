// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should support number type 1`] = `
<input
  className="styles__c-input-text___3TAv1 styles__c-input-text--large___28EaR"
  onChange={[MockFunction]}
  type="number"
  value="42"
/>
`;
